programs/lz4cli.c

```diff
@@ -49,7 +49,7 @@
 #  pragma warning(disable : 4127)      /* disable: C4127: conditional expression is constant */
 #endif
 
-#define _POSIX_SOURCE 1        /* for fileno() within <stdio.h> on unix */
+#define _XOPEN_VERSION 600 /* POSIX.2001, for fileno() within <stdio.h> on unix */
```

Comment:
@Cyan4973 - yes, I can, the source is included in the commit message:

    request POSIX.2001 definitions for fileno()
    
    this also fixes compilation on Solaris 10
    
    cf. test_macros(7) Linux man page:
    
    > _POSIX_SOURCE
    >   Defining this obsolete macro with any value is equivalent to  defin‐
    >   ing _POSIX_C_SOURCE with the value 1.
    >
    >   Since  this macro is obsolete, its usage is generally not documented
    >   when discussing feature test macro requirements in the man pages.